cupy/cuda/memory.pyx

```diff
@@ -1973,3 +1973,22 @@ cdef class PythonFunctionAllocator:
             size, self._malloc_func,
             self._free_func, device.get_device_id())
         return MemoryPointer(mem, 0)
+
+
+cdef dict _allocations = dict()
+
+
+cdef public void* cupy_malloc_ext(
+        ssize_t size, int device, void* stream) with gil:
+    # TODO(ecastill) dynamically change stream & device?
+    cdef curr_stream = stream_module.get_current_stream().ptr
+    if <intptr_t>stream != <intptr_t>curr_stream:
+        raise RuntimeError(f'stream must be the current one')
+    mem = alloc(size)
+    _allocations[mem.ptr] = mem
+    return <void *>mem.ptr
```

Comment:
This is safe because `Memory.ptr` is a cdef class attribute?

Reply:
I hope so!

Reply:
`alloc()` returns `MemoryPointer` which have `readonly intptr_t ptr`, so this should be OK :)